src/main/java/org/sunbird/workflow/service/impl/WorkflowServiceImplV2.java

```diff
@@ -1,42 +1,354 @@
+import org.sunbird.workflow.models.WfAction;
 import org.sunbird.workflow.models.WfRequest;
+import org.sunbird.workflow.postgres.entity.WfStatusEntityV2;
+import org.sunbird.workflow.postgres.repo.WfAuditRepo;
+import org.sunbird.workflow.postgres.repo.WfStatusRepoV2;
+import org.sunbird.workflow.producer.Producer;
+import org.sunbird.workflow.service.UserProfileWfService;
 import org.sunbird.workflow.service.WorkflowServiceV2;
+import org.sunbird.workflow.util.RequestInterceptor;
+
+import java.io.IOException;
+import java.util.*;
 
 @Service
 public class WorkflowServiceImplV2 implements WorkflowServiceV2 {
 
-	@Override
-	public Response workflowTransition(String userToken, WfRequest wfRequest) {
-
-		return null;
-	}
-
-	@Override
-	public Response getWfApplication(String userToken, String wfId) {
-		// TODO Auto-generated method stub
-		return null;
-	}
-
-	@Override
-	public Response wfApplicationSearch(String userToken, SearchCriteria criteria) {
-		// TODO Auto-generated method stub
-		return null;
-	}
-
-	private String validateRequest(WfRequest wfRequest) {
-		String errorMsg = "";
-
-		if (StringUtils.isEmpty(wfRequest.getWfId())) {
-			// Status should be initial.
-		} else {
-			// Status must not be initial.
-		}
-		// check for other parameters based on requirement.
-		return errorMsg;
-	}
+    @Autowired
+    private WfStatusRepoV2 wfStatusRepoV2;
+
+    @Autowired
+    private WfAuditRepo wfAuditRepo;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private Configuration configuration;
+
+    @Autowired
+    private UserProfileWfService userProfileWfService;
+
+    @Autowired
+    private RequestServiceImpl requestServiceImpl;
+
+    @Autowired
+    private Producer producer;
+
+    @Autowired
+    RequestInterceptor requestInterceptor;
+    Logger log = LogManager.getLogger(WorkflowServiceImplV2.class);
+
+
+    /**
+     * Change the status of workflow application
+     *
+     * @param userToken
+     * @param wfRequest
+     * @return
+     */
+    @Override
+    public Response workflowTransition(String userToken, WfRequest wfRequest) {
+        Response response = new Response();
+        String changedStatus = null;
+        String userId = validateAuthTokenAndFetchUserId(userToken);
+        if (!CollectionUtils.isEmpty(wfRequest.getUpdateFieldValues())) {
+            List<HashMap<String, Object>> validTerms = new ArrayList<>();
+            ArrayList<String> invalidTerms = new ArrayList<>();
+            for (HashMap<String, Object> updatedField : wfRequest.getUpdateFieldValues()) {
+                if (StringUtils.isEmpty((String) updatedField.get(Constants.APPROVAL_STATUS)) || updatedField.get(Constants.APPROVAL_STATUS).equals(Constants.DRAFT)) {
+                    validTerms.add(updatedField);
+                } else {
+                    invalidTerms.add((String) updatedField.get(Constants.IDENTIFIER));
+                }
+            }
```

Comment:
This part of code needs to be moved to validateRequest method

Reply:
moved this part into validate method